### PR TITLE
Move Abi V0 files to `starknet_py/abi/v0`

### LINKED ISSUE #1270: Move Abi V0 files to `starknet_py/abi/v0` directory

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #1270 

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/guide/serialization.rst [hunk 1]
```diff
@@ -88,9 +88,10 @@ For most use cases using high level :obj:`Contract <starknet_py.contract.Contrac
 and deserialization for you. If you need more flexibility you can use lower level serialization API, see :ref:`Serializers`
 for more details.
 
-:obj:`AbiParser <starknet_py.net.models.abi.parser.AbiParser>` transforms ABI into
-:obj:`Abi dataclass <starknet_py.net.models.abi.model.Abi>` that can be used for creating serializers. This way you can
-easily deserialize events or serialize function's inputs.
+:obj:`AbiParser <starknet_py.net.models.abi.v0.parser.AbiParser>` transforms ABI into
+:obj:`Abi dataclass <starknet_py.net.models.abi.v0.model.Abi>` that can be used for creating serializers. This way you can
```
Comment (ddoktorski) [inline docs/guide/serialization.rst:92]: The paths are incorrect (they were also incorrect before this change)
```suggestion
:obj:`AbiParser <starknet_py.abi.v2.parser.AbiParser>` transforms ABI into
:obj:`Abi class <starknet_py.abi.v2.model.Abi>` that can be used for creating serializers. This way you can
```

File: docs/api/abi.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@ Abi
 
 Module containing representation of contract abi and parser for creating it from parsed json.
 
-.. py:module:: starknet_py.abi
+.. py:module:: starknet_py.abi.v2
```
Comment (ddoktorski) [inline docs/api/abi.rst:6]: Could we also include `AbiParser` and `Abi` documentation for `v0` and `v1` at the bottom of this file?

File: starknet_py/serialization/factory.py [hunk 3]
```diff
@@ -3,9 +3,9 @@
 from collections import OrderedDict
 from typing import Dict, List, Union
 
-from starknet_py.abi.model import Abi
-from starknet_py.abi.v1.model import Abi as AbiV1
-from starknet_py.abi.v2.model import Abi as AbiV2
+from starknet_py.abi.v0 import Abi
```
Comment (ddoktorski) [inline starknet_py/serialization/factory.py:6]: nit: Also in other files
```suggestion
from starknet_py.abi.v0 import Abi as AbiV0
```

File: starknet_py/serialization/serialization_test.py [hunk 4]
```diff
@@ -4,9 +4,9 @@
 import json
 from typing import NamedTuple
 
-from starknet_py.abi.parser import AbiParser
-from starknet_py.abi.v1.parser import AbiParser as AbiParserV1
-from starknet_py.abi.v2.parser import AbiParser as AbiParserV2
+from starknet_py.abi.v0 import AbiParser
```
Comment (ddoktorski) [inline starknet_py/serialization/serialization_test.py:7]: nit: Also in other files
```suggestion
from starknet_py.abi.v0 import AbiParser as AbiParserV0
```

File: docs/migration_guide.rst [hunk 5]
```diff
@@ -9,6 +9,7 @@ Migration guide
 -----------------------
 
 1. :class:`StarknetEthProxyCheck` has been removed from the Proxy checks
+2. ``starknet_py.abi`` has been moved to ``starknet_py.abi.v0``
```
Comment (ddoktorski) [inline docs/migration_guide.rst:12]:
```suggestion
2. In :ref:`Abi` module the code related to Cairo 0 has been moved from ``starknet_py.abi`` to ``starknet_py.abi.v0``
```

**Review by ddoktorski — CHANGES_REQUESTED** [review]

File: starknet_py/serialization/factory.py (hunk 3 above)
Comment (tkumor3) [reply starknet_py/serialization/factory.py:6]: I will add for files where we import more than one `starknet_py.abi.*`

**Review by ddoktorski — APPROVED** [review]

**Pull request merged**